src/components/Counter.js

```diff
@@ -1,42 +1,53 @@
 import React, { Component } from "react";
-import { connect } from 'react-redux';
-import { increment, decrement } from '../actions';
+import { connect } from "react-redux";
+import { increment, decrement } from "../actions";
 
 class Counter extends Component {
-    incrementIfOdd = () => {
-        // Stretch Problem: Implement an increment function that
-        // only increments if the counter value is odd
-    };
+  incrementIfOdd = () => {
+    // Stretch Problem: Implement an increment function that
+    // only increments if the counter value is odd
 
-    incrementAsync = () => {
-        // Stretch Problem: Implement an increment function that
-        // increments after waiting for one second
-    };
+    if (this.props.count % 2 === 1) {
+      return this.props.increment();
+    }
+  };
+
+  incrementAsync = () => {
+    // Stretch Problem: Implement an increment function that
+    // increments after waiting for one second
+  };
 
-    render() {
-        // Fill in the two button onClick methods
-        // Upon clicking these buttons, the count
-        // should decrement or increment accordingly
-        return (
-            <p>
-                Clicked: {this.props.count} times
-                <button onClick={() => {/* Fill me in */ }}>
-                    +
-                </button>
-                <button onClick={() => {/* Fill me in */ }}>
-                    -
-                </button>
-                 {/* Uncomment these button tags if you got
+  render() {
+    // Fill in the two button onClick methods
+    // Upon clicking these buttons, the count
+    // should decrement or increment accordingly
+    const { count, increment, decrement } = this.props;
```

Comment:
excellent use of destructing here making the rest of your code more clean and semantic.